IopHomeNode.proto3

```diff
@@ -524,15 +524,16 @@ message ListHomeNodePlansResponse {
 
 // Description of a home node hosting plan.
 message HomeNodePlan {
-  bytes planId = 1;                                // GUID of the plan. If a node wants to change the plan, it has to change the GUID as well,
-                                                   // thus effectively creating a new plan.
-  string identityType = 2;                         // Type of identity this plan is created for. This can include "*" as a wildcard character
+  bytes planId = 1;                                // Identifier of the plan is an SHA1 hash of the HomeNodePlan structure in Google Protobuf v3 encoding 
+                                                   // with planId initialized to a byte array of 20 zero bytes.
```

Comment:
The presentation layer could have a "hash" property, and application could use it. I don't really see it's a problem. Since upper layer is aware of its lower layer, but not vise versa. Like TCP application is aware of its TCP configurations.